fe/fe-core/src/main/java/com/starrocks/connector/iceberg/hadoop/IcebergHadoopCatalog.java

```diff
@@ -0,0 +1,200 @@
+import org.apache.iceberg.CatalogProperties;
+import org.apache.iceberg.CatalogUtil;
+import org.apache.iceberg.PartitionSpec;
+import org.apache.iceberg.Schema;
+import org.apache.iceberg.Table;
+import org.apache.iceberg.catalog.Namespace;
+import org.apache.iceberg.catalog.TableIdentifier;
+import org.apache.iceberg.hadoop.HadoopCatalog;
+import org.apache.logging.log4j.LogManager;
+import org.apache.logging.log4j.Logger;
+
+import java.net.URI;
+import java.util.ArrayList;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.stream.Collectors;
+
+import static com.starrocks.connector.ConnectorTableId.CONNECTOR_ID_GENERATOR;
+import static com.starrocks.connector.iceberg.IcebergConnector.ICEBERG_CUSTOM_PROPERTIES_PREFIX;
+import static org.apache.iceberg.CatalogProperties.WAREHOUSE_LOCATION;
+
+public class IcebergHadoopCatalog implements IcebergCatalog {
+    private static final Logger LOG = LogManager.getLogger(IcebergHadoopCatalog.class);
+    public static final String LOCATION_PROPERTY = "location";
+
+    private final Configuration conf;
+    private final HadoopCatalog delegate;
+
+    public IcebergHadoopCatalog(String name, Configuration conf, Map<String, String> properties) {
+        this.conf = conf;
+        Map<String, String> copiedProperties = Maps.newHashMap(properties);
+
+        properties.forEach((key, value) -> {
+            String newKey = key.toLowerCase();
+            if (newKey.startsWith(ICEBERG_CUSTOM_PROPERTIES_PREFIX)) {
+                newKey = newKey.substring(ICEBERG_CUSTOM_PROPERTIES_PREFIX.length());
+                copiedProperties.put(newKey, value);
+            }
+        });
+
+        copiedProperties.put(CatalogProperties.FILE_IO_IMPL, IcebergCachingFileIO.class.getName());
+        copiedProperties.put(CatalogProperties.METRICS_REPORTER_IMPL, IcebergMetricsReporter.class.getName());
+
+        if (!copiedProperties.containsKey(WAREHOUSE_LOCATION)) {
+            throw new IllegalArgumentException(String.format("Iceberg hadoop catalog must set warehouse location (\"%s\" = \"s3://path/to/warehouse\").",
+                    ICEBERG_CUSTOM_PROPERTIES_PREFIX + WAREHOUSE_LOCATION));
+        }
+
+        delegate = (HadoopCatalog) CatalogUtil.loadCatalog(HadoopCatalog.class.getName(), name, copiedProperties, conf);
+    }
+
+    @Override
+    public IcebergCatalogType getIcebergCatalogType() {
+        return IcebergCatalogType.HADOOP_CATALOG;
+    }
+
+    @Override
+    public Table getTable(String dbName, String tableName) throws StarRocksConnectorException {
+        return delegate.loadTable(TableIdentifier.of(dbName, tableName));
+    }
+
+    @Override
+    public List<String> listAllDatabases() {
+        return delegate.listNamespaces().stream()
+                .map(ns -> ns.level(0))
+                .collect(Collectors.toList());
+    }
+
+    @Override
+    public void createDb(String dbName, Map<String, String> properties) {
```

Comment:
If most of the catalog code is the same, we'd better abstract these methods into the interface to avoid copying it again.

Reply:
we can do it in the future